### ISSUE TITLE
Block: Math - Keyboard trap in editor

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#73447]
### Description

Not possible to leave the textarea from the keyboard in the math editor.

### Step-by-step reproduction instructions

1. Add a math block
2. Place your cursor inside the editor textarea.
3. Use `tab`, `shift+tab` and `esc` to attempt to leave the editor field.

There needs to be a mechanism to exit the editor. If tabbing is constrained while editing, then there should be an explicit close button that can be tabbed to; and esc should close the editor.

### Environment info

-  WP 6.9-RC2-61266, Gutenberg 22.1.0
-  Chrome & Firefox, Windows 11

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
